indigo_app/static/javascript/indigo/views/document_editor.js

```diff
@@ -388,32 +379,35 @@
     },
 
     editTable: function(e) {
-      this.editActivityStarted('table');
-      var $btn = $(e.currentTarget),
-          table = document.getElementById($btn.data('table-id'));
-
-      if (!this.tableEditor.canEditTable(table)) {
-        alert($t('This table contains content that cannot be edited in this mode. Edit the table in text mode instead.'));
-      } else {
-        this.tableEditor.editTable(table);
-        // disable other table edit buttons
-        this.$('.edit-table').prop('disabled', true);
+      const tableId = e.currentTarget.dataset.tableId;
+
+      if (this.confirmAndDiscardChanges()) {
```

Comment:
where else is this called that could potentially change the DOM, but we're not waiting for it to settle?

Reply:
it only applies in this case because we need the updated XML to be re-rendered to HTML, because table editing happens directly the html.